app-engine/java/src/com/google/cloud/tools/intellij/appengine/java/cloud/GradlePluginDeploymentSourceType.java

```diff
@@ -59,7 +61,12 @@ public GradlePluginDeploymentSourceType() {
   @Override
   protected BeforeRunTask createBuildTask(Module module) {
     ExternalSystemTaskExecutionSettings settings = new ExternalSystemTaskExecutionSettings();
-    settings.setExternalProjectPath(module.getModuleFilePath());
+
+    Optional<String> gradleModuleDirOptional =
+        AppEngineStandardGradleModuleComponent.getInstance(module).getGradleModuleDir();
+
+    gradleModuleDirOptional.ifPresent(settings::setExternalProjectPath);
```

Comment:
What would happen if this module dir is not available? Will this task still be valid?

Reply:
the task will still be present, just with no gradle project on which to run the task. This should never happen in normal operation (since the deployment source would not have been created in the first place).

I should add some logging though.

Reply:
done